src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+    private int halfOfSize = size / 2;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node currentNode = new Node<>(null, value, null);
+            tail = currentNode;
+            head = currentNode;
+            size++;
+        } else if (size > 0) {
+            Node previousNode = head;
+            Node currentNode = new Node(previousNode, value, null);
+            previousNode.next = currentNode;
+            head = currentNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Your index was: " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0 && size > 0) {
+            Node newNode = new Node<>(null, value, tail);
+            tail.prev = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+        int counter = 0;
+        Node currentNode = tail;
+        if (index >= halfOfSize) {
+            counter = size - 1;
+            currentNode = head;
+        }
+        if (index >= 0 && index < size) {
+            while (currentNode != null) {
+                if (counter == index) {
+                    Node previousNode = currentNode.prev;
+                    Node addedNode = new Node(previousNode, value, currentNode);
+                    if (previousNode != null) {
+                        previousNode.next = addedNode;
+                    }
+                    currentNode.prev = addedNode;
+                    size++;
+                }
+                if (size < halfOfSize) {
+                    counter++;
+                    currentNode = currentNode.next;
+                } else if (size >= halfOfSize) {
+                    counter--;
+                    currentNode = currentNode.prev;
+                }
+            }
+        }
```

Comment:
This method is too big. 
How about extracting this code to two separate methods: `getByIndexForward()` and `getByIndexBackward()`?